apps/test/unit/code-studio/pd/form_components_func/FormControllerTest.js

```diff
@@ -56,71 +56,60 @@ describe('FormController', () => {
       };
     });
 
-    const getCurrentPage = () => form.find('Pagination').prop('activePage');
-    const getData = page => form.find(page).prop('data');
-    const getErrors = page => form.find(page).prop('errors');
+    const getCurrentPage = () => form.findOne('Pagination').props.activePage;
+    const getData = page => form.findOne(page).props.data;
+    const getErrors = page => form.findOne(page).props.errors;
     const setPage = i => {
-      form.find('Pagination').prop('onSelect')(i + 1);
-      form.update();
+      form.findOne('Pagination').props.onSelect(i + 1);
     };
 
     it('Initially renders the first page', () => {
-      form = mount(<FormController {...defaultProps} />);
+      form = isolateComponent(<FormController {...defaultProps} />);
       expect(getCurrentPage()).to.equal(1);
-      expect(form.find(DummyPage1)).to.have.length(1);
-      expect(form.find(DummyPage2)).to.have.length(0);
-      expect(form.find(DummyPage3)).to.have.length(0);
+      expect(form.exists(DummyPage1));
+      expect(!form.exists(DummyPage2));
+      expect(!form.exists(DummyPage3));
     });
 
-    it('Displays page buttons on each page', () => {
```

Comment:
this was testing functionality of the Pagination component

Reply:
does isolate-component actually prevent you from testing the contents of the Pagination component? just checking my understanding that it is actually more like `shallow` than `mount` in that way.

Reply:
yes, it is more like shallow, it couldn't find the PaginationButton components in the tree